src/goboy.go

```diff
@@ -1,7 +1,66 @@
-package main
+package GoBoy
 
-type GameBoy struct{
+type GameBoy struct {
     mram [8 * 1024]uint8
     vram [8 * 1024]uint8
 }
 
+type Register struct {
+    regs [6]uint16
+}
+
+func (r *Register) GetA() uint8 {
+    return uint8(r.regs[0] >> 8)
+}
+
+func (r *Register) GetF() uint8 {
+    return uint8(r.regs[0])
+}
+
+func (r *Register) GetB() uint8 {
+    return uint8(r.regs[1] >> 8)
+}
+
+func (r *Register) GetC() uint8 {
+    return uint8(r.regs[1])
+}
+
+func (r *Register) GetD() uint8 {
+    return uint8(r.regs[2] >> 8)
+}
+
+func (r *Register) GetE() uint8 {
+    return uint8(r.regs[2])
+}
+
+func (r *Register) getH() uint8 {
```

Comment:
Any reason we switch casing here? I actually prefer it this way, not with "get" capitalized

Reply:
My mistake. Apparently, in go, caps attributes are public and lowercase are private.
https://stackoverflow.com/questions/20850047/public-private-upper-case-lower-case